src/Layout/Page/Page.component.tsx

```diff
@@ -2,50 +2,215 @@
+};
+
+// Types
+// ------------------------------------------------------------------------------------------------------------------
+type SidebarAlignment = 'left' | 'right';
+
+// Styled Components
+// ------------------------------------------------------------------------------------------------------------------
+const StyledPage = styled.section`
     display: flex;
     flex-direction: column;
     height: 100vh;
-    max-width: 100vw;
-
-    header {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+    max-width: 100%;
+`;
 
-    footer {
-        flex: 0 0 auto;
-        max-width: 100vw;
-    }
+const StyledLayout = styled.section<StyledLayoutProps>`
+    max-width: ${props => props.layoutWidth};
+    background-color: ${props => props.layoutBackgroundColor};
+    margin: 0 auto;
 `;
 
-const StyledBodyElement = styled.section`
-    flex: 1 0 auto;
-    background-color: ${colors.silver.base};
-    max-width: 100vw;
-    align-self: stretch;
-    padding: ${sizes.padding.lg} ${sizes.padding.xl};
+const StyledContent = styled.div<StyledContentProps>`
+    max-width: ${props => props.contentWidth};
+    background-color: ${props => props.contentBackgroundColor};
+    margin: 0 auto;
 `;
 
+// Interfaces
+// ------------------------------------------------------------------------------------------------------------------
 interface PageProps {
+    /** The header of the website. This can be a component. */
+    header?: any;
+    /** The footer of the website. This can be a component. */
+    footer?: any;
+    /** The height of the header. It will not hide on overflow. */
+    headerHeight?: string | number;
+    /** The height of the footer. It will not hide on overflow. */
+    footerHeight?: string | number;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface DefaultLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
+    /** The background color of the content within the layout area. Default is transparent. */
+    contentBackgroundColor?: string;
+    /** Additional classname. */
+    className?: string;
+    children?: any;
+}
+
+interface SidebarLayoutProps {
+    /** The width of the layout, edge to edge. */
+    layoutWidth?: string | number;
+    /** The width of the content within the layout. */
+    contentWidth?: string | number;
+    /** The width of the area for the sidebar. */
+    sidebarWidth?: string | number;
+    /** If the sidebar is on the left or the right of the layout */
+    sidebarAlign?: SidebarAlignment;
+    /** The actual sidebar. This can be a component. */
+    sidebar?: any;
+    /** The background color of the layout. Default is transparent. */
+    layoutBackgroundColor?: string;
```

Comment:
This prop and the subsequent one can be a slippery slope. I'm kosher leaving them here, but the second I start hearing about needing font-color, anchor tag treatment, etc. overrides, I'll yank these props out of here and ask people to extend this using classes.

Reply:
For some of those we can create a `Hero` component that functions like Bootstrap's Hero, but this also solves that nicely. I just wanna get ahead the 'onslaught', but I don't think that merits a change to this. It's a nice feature to have.